openpype/hosts/maya/tools/mayalookassigner/arnold_standin.py

```diff
@@ -0,0 +1,235 @@
+import os
+import json
+from collections import defaultdict
+import logging
+
+from maya import cmds
+
+from openpype.pipeline import legacy_io
+from openpype.client import get_last_version_by_subset_name
+from openpype.hosts.maya import api
+from . import lib
+
+
+log = logging.getLogger(__name__)
+
+
+ATTRIBUTE_MAPPING = {
+    "primaryVisibility": "visibility",  # Camera
+    "castsShadows": "visibility",  # Shadow
+    "receiveShadows": "receive_shadows",
+    "aiSelfShadows": "self_shadows",
+    "aiOpaque": "opaque",
+    "aiMatte": "matte",
+    "aiVisibleInDiffuseTransmission": "visibility",
+    "aiVisibleInSpecularTransmission": "visibility",
+    "aiVisibleInVolume": "visibility",
+    "aiVisibleInDiffuseReflection": "visibility",
+    "aiVisibleInSpecularReflection": "visibility",
+    "aiSubdivUvSmoothing": "subdiv_uv_smoothing",
+    "aiDispHeight": "disp_height",
+    "aiDispPadding": "disp_padding",
+    "aiDispZeroValue": "disp_zero_value",
+    "aiStepSize": "step_size",
+    "aiVolumePadding": "volume_padding",
+    "aiSubdivType": "subdiv_type",
+    "aiSubdivIterations": "subdiv_iterations"
+}
+
+
+def calculate_visibility_mask(attributes):
+    # https://arnoldsupport.com/2018/11/21/backdoor-setting-visibility/
+    mapping = {
+        "primaryVisibility": 1,  # Camera
+        "castsShadows": 2,  # Shadow
+        "aiVisibleInDiffuseTransmission": 4,
+        "aiVisibleInSpecularTransmission": 8,
+        "aiVisibleInVolume": 16,
+        "aiVisibleInDiffuseReflection": 32,
+        "aiVisibleInSpecularReflection": 64
+    }
+    mask = 255
+    for attr, value in mapping.items():
+        if attributes.get(attr, True):
+            continue
+
+        mask -= value
+
+    return mask
+
+
+def get_nodes_by_id(standin):
+    """Get node id from aiStandIn via json sidecar.
+
+    Args:
+        standin (string): aiStandIn node.
+
+    Returns:
+        (dict): Dictionary with node full name/path and id.
+    """
+    path = cmds.getAttr(standin + ".dso")
+    json_path = None
+    for f in os.listdir(os.path.dirname(path)):
+        if f.endswith(".json"):
+            json_path = os.path.join(os.path.dirname(path), f)
+            break
+
+    with open(json_path, "r") as f:
+        return json.load(f)
+
+
+def shading_engine_assignments(shading_engine, attribute, nodes, assignments):
```

Comment:
I feel like this might need a docstring. This likely came from elsewhere but I'm not entirely sure what this function does by a quick read of it.

Reply:
Yeah, fair.